internal/pgbackrest/pgbackrest.go

```diff
@@ -60,22 +62,44 @@ func (exec Executor) StanzaCreateOrUpgrade(ctx context.Context, configHash strin
 		stanzaCmd = "upgrade"
 	}
 
+	var reposWithVolumes []v1beta1.PGBackRestRepo
+	for _, repo := range postgresCluster.Spec.Backups.PGBackRest.Repos {
+		if repo.Volume != nil {
+			reposWithVolumes = append(reposWithVolumes, repo)
+		}
+	}
+
+	grep := "grep %s-path /etc/pgbackrest/conf.d/pgbackrest_instance.conf"
```

Comment:
@andrewlecuyer If we checked that the number of repos in the container config doesn't equal what's in the spec, wouldn't we get the added benefit of handling cases where a repo is removed (regardless of the type of repo)? The hash would still cover the cases where number of repos stays the same, but endpoint changes, right?